sync2/tokens_table.go

```diff
@@ -42,17 +44,32 @@ func NewTokensTable(db *sqlx.DB, secret string) *TokensTable {
 		-- TODO: FK constraints to devices table?
 		user_id TEXT NOT NULL,
 		device_id TEXT NOT NULL,
+		expired BOOL NOT NULL DEFAULT FALSE,
 		last_seen TIMESTAMP WITH TIME ZONE NOT NULL
 	);`)
 
 	// derive the key from the secret
 	hash := sha256.New()
 	hash.Write([]byte(secret))
 
-	return &TokensTable{
+	t := &TokensTable{
 		db:     db,
 		key256: hash.Sum(nil),
 	}
+
+	var deleteExpiredTokens func()
+	deleteExpiredTokens = func() {
+		deleted, err := t.deleteExpiredTokensAfter(30 * 24 * time.Hour)
+		if err != nil {
+			logger.Warn().Err(err).Msg("failed to delete expired tokens")
+			return
+		}
+		logger.Trace().Int64("deleted", deleted).Msg("deleted expired tokens")
+		time.AfterFunc(time.Hour, deleteExpiredTokens)
+	}
+	time.AfterFunc(time.Hour, deleteExpiredTokens)
```

Comment:
Please don't tie this to the `New...` constructor. This is going to be such a faff to work with in tests otherwise.